scripts/inner/kubectl-kubestellar-ensure-wmw

```diff
@@ -98,25 +79,27 @@ esac
 
 set -e
 
-in_cluster=
-if $IN_CLUSTER; then
-    in_cluster="--in-cluster"
+if [ -z "$wds_kubeconfig" ]; then
+    wds_kubeconfig="$(mktemp "${wmw_name}.kubeconfig#XXXX")"
+    kc_is_temp=true
 fi
 
-space_config="${KUBECONFIG_DIR}/${PROVIDER_NAMESPACE}-${wmw_name}"
-kubectl-kubestellar-create-space --space-name $wmw_name --provider-name $PROVIDER_NAME --sm-core-config $SM_CONFIG
-kubectl-kubestellar-get-config-for-space --space-name $wmw_name --sm-core-config $SM_CONFIG --sm-context $SM_CONTEXT ${in_cluster} --output $space_config
+kubectl-kubestellar-space-ensure "${sub_flags[@]}" "${kubectl_flags[@]}" $wmw_name --output-kubeconfig "$wds_kubeconfig"
 
-KUBECONFIG=$SM_CONFIG kubestellar-kube-bind "$wmw_name" "edgeplacements"
-KUBECONFIG=$SM_CONFIG kubestellar-kube-bind "$wmw_name" "customizers"
-KUBECONFIG=$SM_CONFIG kubestellar-kube-bind "$wmw_name" "singleplacementslices"
+kubestellar-kube-bind "${sub_flags[@]}" "${kubectl_flags[@]}" "$wmw_name" "edgeplacements"
+kubestellar-kube-bind "${sub_flags[@]}" "${kubectl_flags[@]}" "$wmw_name" "customizers"
+kubestellar-kube-bind "${sub_flags[@]}" "${kubectl_flags[@]}" "$wmw_name" "singleplacementslices"
 
 function reconcile_kube_CRDs() {
 	if [ "$want_kube" == true ]; then
-		kubectl --kubeconfig $space_config "${kubectl_flags[@]}" apply --server-side=true -R -f "$bindir/../config/kube/crds/"
+		kubectl --kubeconfig "$wds_kubeconfig" apply --server-side=true -R -f "$bindir/../config/kube/crds/"
 	else
-		kubectl --kubeconfig $space_config "${kubectl_flags[@]}" delete -R -f "$bindir/../config/kube/crds/"
+		kubectl --kubeconfig "$wds_kubeconfig" delete -R -f "$bindir/../config/kube/crds/"
 	fi
 }
 
 reconcile_kube_CRDs
+
+if [ "$kc_is_temp" == true ]
```

Comment:
Perhaps switch to using trap at the point where you define wds_kubeconfig?  I learned this from Jun:
```
wds_kubeconfig="$(mktemp "${wmw_name}.kubeconfig#XXXX")"
trap "rm -f $wds_kubeconfig" EXIT
```

Reply:
I deliberately did _not_ use `trap` in this case, in order to leave some evidence behind to aid in investigation in the failure case.